### ISSUE TITLE
Some questions about metrics and evaluations

### ISSUE THREAD

**Issue by ScuLancer** [DiffusionLight/DiffusionLight#44]
Hi,
I am currently testing your code on the HDRMaps_4K dataset using your pre-trained weights. I have a few queries regarding the pipeline

1.**Input Preprocessing:** Could you provide more details on how the original 4K environment maps are processed into the 60° FOV LDR (1024x1024) inputs? I noticed that the values in the Laval Indoor dataset are very low (appearing dark), while Poly Haven seems normal.

2.**Evaluation Protocol:** How exactly are the GT and estimated EXR results handled before calculating metrics? I understand from the paper that they are resized to 128x256 before rendering the spheres for the three materials. What is the final resolution of these rendered spheres?

3.**Metric Discrepancy:** I ran the inference using your official weights. My evaluation code is listed below. However, my results are quite different from the paper's benchmarks.
Paper: Mirror Material (si-RMSE: 0.60 / Norm-RMSE: 0.43)
My Run: Mirror Material (si-RMSE: 0.23 / Norm-RMSE: 0.26)

<img width="999" height="338" alt="Image" src="https://github.com/user-attachments/assets/98995dd7-ec2d-4567-94d7-44291de355cb" />

<img width="1280" height="525" alt="Image" src="https://github.com/user-attachments/assets/d49be74b-29e3-432c-8d66-26cd64ffe62d" />

I am confused by this large difference. Are there any specific post-processing steps I might be missing? I want to ensure my evaluation setup is consistent with yours.

[Calculate_Metrics_0125.py](https://github.com/user-attachments/files/24890379/Calculate_Metrics_0125.py)

Thanks for your help!

**Comment by pureexe** [comment]
Hi, thank you for your interest in our work.

For evaluation details, please refer to the following repository, which contains the full evaluation pipeline used in our experiments:
https://github.com/DiffusionLight/DiffusionLight-evaluation

Below are answers to your questions.

### 1. Input Preprocessing

We follow the same preprocessing protocol as StyleLight.

We crop the HDR environment map using a 60° horizontal FOV at a resolution of 192×256.

The cropped image is then resized to 1024×1024 with black border padding.

Tone mapping follows StyleLight’s implementation: gamma 2.4, mapping the 99th percentile to 0.9.
Reference code:
https://github.com/Wanggcong/StyleLight/blob/9dcd56992b70c027e2be3708464bbc8403627eea/data_prepare_laval.py#L33

We believe the input images are not excessively dark. For example:

[AG8A4715-42874a83c4.png](https://vistec-my.sharepoint.com/personal/pakkapon_p_s19_vistec_ac_th/_layouts/15/onedrive.aspx?viewid=da3f2183%2De61c%2D42ab%2D80da%2D93d05ea80580&id=%2Fpersonal%2Fpakkapon%5Fp%5Fs19%5Fvistec%5Fac%5Fth%2FDocuments%2Fpublic%2FVLL%2FDiffusionLight%2Frelease%5Fresult%2Ftest%5Fimage%2Findoor%2F1024x1024%2FAG8A4715%2D42874a83c4%2Epng&parent=%2Fpersonal%2Fpakkapon%5Fp%5Fs19%5Fvistec%5Fac%5Fth%2FDocuments%2Fpublic%2FVLL%2FDiffusionLight%2Frelease%5Fresult%2Ftest%5Fimage%2Findoor%2F1024x1024)

<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/76723df0-9b35-46ae-ac8a-11f7a8373402" />

Note: To avoid any preprocessing mismatch, we strongly recommend using the preprocessed inputs we provide directly, rather than re-implementing the preprocessing yourself. This should result in more consistent and accurate evaluation scores.
You can download them here:
https://vistec-my.sharepoint.com/personal/pakkapon_p_s19_vistec_ac_th/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fpakkapon%5Fp%5Fs19%5Fvistec%5Fac%5Fth%2FDocuments%2Fpublic%2FVLL%2FDiffusionLight%2Frelease%5Fresult&ga=1

### 2. Evaluation Protocol

We follow the StyleLight evaluation protocol.
Please see the README in the evaluation repository for step-by-step details:
https://github.com/DiffusionLight/DiffusionLight-evaluation

### 3. Metric Discrepancy

For the metric discrepancy you observed, we recommend:

Running inference using the preprocessed inputs provided in the OneDrive link, and

Evaluating the results using the official evaluation repository linked above.

This should significantly reduce the discrepancy between your results and the numbers reported in the paper.
